cmd/thanos/bucket.go

```diff
@@ -213,4 +217,191 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 			return printBlock(id)
 		})
 	}
+
+	inspect := cmd.Command("inspect", "inspect all blocks in the bucket")
+	selector := inspect.Flag("selector ", "Selects blocks based on label, e.g. -l key1=value1,key2=value2. All key value pairs must match.").Short('l').
```

Comment:
I think it's just a matter of how the command line interface should be. I guess using the same flag ("selector") for both would require some kind of query syntax.

Do you mean selection by the duration of the time range or by the start and end date of the blocks? I think the `journalctl` --since and --until flags are really nice for that. But I think if we implement these flags, we should do it only for a pre-specificied date format and not for strings like "1 hour ago" ([sytemd doc](https://www.freedesktop.org/software/systemd/man/systemd.time.html#)). I couldn't find a golang library for parsing these formats the last time I searched.

Reply:
Yea, pre-specified date will work as well as a start. But can see human readable duration parsing for example here https://github.com/trstringer/go-systemd-time

Anyway, I would add that as improvements to this command, so we can do that later (:

Reply:
Ah nice :). Ok so we'll close this for this PR right now.